### ISSUE TITLE
Exception thrown when converting to int8_t

### ISSUE THREAD

**Issue by ahoarau** [jbeder/yaml-cpp#848]
This code : 

```c++
#include <yaml-cpp/yaml.h>

int main()
{
    YAML::Node node = YAML::Load("[2, -1, 1]");

    for(std::size_t i = 0; i < node.size(); ++i)
        node[i].as<int8_t>();
    return 0;
}
```

throws:
```
terminate called after throwing an instance of 'YAML::TypedBadConversion<signed char>'
  what():  yaml-cpp: error at line 0, column 0: bad conversion
Aborted (core dumped)
```
I'm guessing this is similar to https://github.com/jbeder/yaml-cpp/issues/844

yaml-cpp version used: `0.5.2-4ubuntu1`

**Comment by dota17** [comment]
Fixed by #864.

**Comment by ahoarau** [comment]
Excellent , well done !
